derive/src/lib.rs

```diff
@@ -213,3 +247,180 @@ fn impl_from_args(input: &DeriveInput) -> TokenStream2 {
         }
     }
 }
+
+#[proc_macro_attribute]
+pub fn pyclass(attr: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
nit: could we put each proc_macro in its own submodule? this file is getting kinda big.